pilot/cmd/pilot-agent/options/options.go

```diff
@@ -115,4 +115,7 @@ var (
 	// certSigner is cert signer for workload cert
 	certSigner = env.RegisterStringVar("ISTIO_META_CERT_SIGNER", "",
 		"The cert signer info for workload cert")
+
+	istiodSAN = env.RegisterStringVar("ISTIOD_SAN", "",
+		"Override the ServerName used to validate Istiod certificate. Can be used as an alternative to setting /etc/hosts for VMs - discovery address will be an IP:port")
```

Comment:
should we switch our VM docs/code to use this instead

Reply:
I would wait a bit, let's see how it goes with 'docker' - this mode is convenient if we use a custom launcher that can lookup the LB IP (or the Endpoint IP), and set up all env variables.